tests/core/component.test.js

```diff
@@ -180,6 +182,105 @@ suite('Component', function () {
       });
     });
 
+    test('emits componentchanged for single-prop', function (done) {
+      var el = entityFactory();
+      el.setAttribute('position', {x: 0, y: 0, z: 0});
+      el.addEventListener('componentchanged', function (evt) {
+        if (evt.detail.name !== 'position') { return; }
+        assert.shallowDeepEqual(evt.detail.oldData, {x: 0, y: 0, z: 0});
+        assert.shallowDeepEqual(evt.detail.newData, {x: 1, y: 2, z: 3});
+        assert.equal(evt.detail.name, 'position');
+        assert.ok('id' in evt.detail);
+        done();
+      });
+      setTimeout(() => {
+        el.setAttribute('position', {x: 1, y: 2, z: 3});
+      });
+    });
+
+    test('emits componentchanged for value', function (done) {
+      var el = entityFactory();
+      el.addEventListener('componentchanged', function (evt) {
+        if (evt.detail.name !== 'visible') { return; }
+        assert.shallowDeepEqual(evt.detail.oldData, true);
+        assert.shallowDeepEqual(evt.detail.newData, false);
+        assert.equal(evt.detail.name, 'visible');
+        done();
+      });
+      setTimeout(() => {
```

Comment:
Why do we need a timeout? Can we move `var el = entityFactory();` to a setup method? https://github.com/aframevr/aframe/blob/master/tests/core/a-entity.test.js#L27